### PR TITLE
Abstracted the string utils from the Multiselect for easier reuse and testing

### PR DESCRIPTION
**Pull request by jimmynotjim**
Abstracted the string utils from the Multiselect for easier reuse and testing

## Changes

- Renamed _regExpEscape to stringEscape and added tests
- Renamed _filterContains to stringMatch and added tests

## Testing

- `gulp test:unit:scripts`
- `gulp build` and navigate to `/blog` and use the filters (~~the expandable is broken, @sebworks is working on a fix, you'll need to inspect and force it open~~ it's fixed)

## Review

- @anselmbradford 
- @sebworks 
- @KimberlyMunoz 

## Notes

__Before__

```
=============================== Coverage summary ===============================
Statements   : 29.04% ( 778/2679 )
Branches     : 24.25% ( 210/866 )
Functions    : 22.43% ( 96/428 )
Lines        : 29.19% ( 775/2655 )
================================================================================
```

__After__

```
=============================== Coverage summary ===============================
Statements   : 29.13% ( 781/2681 )
Branches     : 24.25% ( 210/866 )
Functions    : 22.9% ( 98/428 )
Lines        : 29.28% ( 778/2657 )
================================================================================
```

`/utils` are now at 79% coverage, this is huge for us!

screen shot 2016-03-25 at 9 50 23 am

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by KimberlyMunoz** [comment]
:+1: I'd add another test comparing strings with the same content but different casing, but it's not necessary.

File: test/unit_tests/modules/util/strings-spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+var BASE_JS_PATH = '../../../../cfgov/unprocessed/js/';
+var ERROR_MESSAGES = require( BASE_JS_PATH + 'config/error-messages-config' );
+var strings = require( BASE_JS_PATH + 'modules/util/strings' );
+var string;
+var control;
+
+describe( 'Strings stringEscape()', function() {
+
+  it( 'should escape a url', function() {
+    string = 'http://google.com'
```
Comment (anselmbradford) [inline test/unit_tests/modules/util/strings-spec.js]: Add trailing `;` to satisfy linter.

File: test/unit_tests/modules/util/strings-spec.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    expect( strings.stringEscape( string ) )
+      .to.equal( 'http://google\\.com' );
+  } );
+
+  it( 'should escape a hyphenated name', function() {
+    string = 'Miller-Webster'
```
Comment (anselmbradford) [inline test/unit_tests/modules/util/strings-spec.js]: Add trailing `;` to satisfy linter.

File: cfgov/unprocessed/js/modules/util/strings.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * @returns {string}   The escaped string
+ */
+function stringEscape( s ) {
+  return s.replace( /[-\\^$*+?.()|[\]{}]/g, '\\$&' );
+}
+
+/**
+ * Tests whether a string matches another
+ * @param   {string}  x  The control string
+ * @param   {string}  y  The comparrison string
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/modules/util/strings.js]: `comparrison`
pirate

Comment (jimmynotjim) [inline cfgov/unprocessed/js/modules/util/strings.js]: I am the worst with typos, but worth it for that gif. Fixed

**Comment by sebworks** [comment]
:+1:

File: test/unit_tests/modules/util/strings-spec.js [hunk 4]
```diff
@@ -40,6 +40,16 @@ describe( 'Strings stringMatch()', function() {
       .to.be.true;
   } );
 
+  it( 'should return true when testing matching strings with differing casing',
```
Comment (KimberlyMunoz) [inline test/unit_tests/modules/util/strings-spec.js]: Thanks! I wasn't sure what the behavior was going to be, so this doubles as documentation. :+1:

**Pull request merged**